### PR TITLE
Ego 291 group controller tests

### PR DESCRIPTION
**Pull request by rtisma**
Related to #291 

- fixed swagger `@ApiImplicitParams` for all endpoints
- renamed methods to be easier to understand. I.e `getGroupsApplications` -> `getApplicationsForGroup`
- implemented `addUsersToGroup` and `deleteUsersFromGroup` endpoints properly for GroupController
- implemented `deleteGroupFromUser` and `addGroupToUser` endpoints properly for UserController
- Transformed `@ManyToMany` User-Group relationship to a `@OneToMany` relationship with a link entity called `UserGroup` using composite ids. This allows for the relationship to be controlled by either side of the association.
- implemented dynamic loading of the entity graph. `AbstractSpecificationBuilder` is used to build the entity graph dynamically and then run the query on it. This eliminates the need for the `@EntityGraph` annotation which is statically defined
- Added new method to `BaseService` interface called `getWithRelationships` which fetches the entity by id with ALL relationships (i.e fully loaded)
- Implemented correct deletion of Groups and Users, by firstly disassociating ALL entities, and then deleting
- Added `EntityServices` utility class, to aid using Service level methods while avoiding Cyclical Dependency hell
- added `spring.jackson.deserialization.FAIL_ON_UNKNOWN_PROPERTIES: true` to return a 400 bad request when the user submits unknown or extra fields in the payload. 
- added logging flag to WebResources and its enabled by default
- Added very detailed and exhaustive GroupControllerTests, testing many to many, one to many relationships as well as other controller based features
- Added fluent functionality to WebResources, to write cleaner tests with slightly less code

### CODE REVIEW

File: src/main/java/bio/overture/ego/model/entity/Group.java [hunk 1]
```diff
@@ -137,14 +117,12 @@
...
-      fetch = FetchType.LAZY,
-      cascade = {CascadeType.PERSIST, CascadeType.MERGE})
-  @JoinTable(
-      name = Tables.GROUP_USER,
-      joinColumns = {@JoinColumn(name = SqlFields.GROUPID_JOIN)},
-      inverseJoinColumns = {@JoinColumn(name = SqlFields.USERID_JOIN)})
   @JsonIgnore
   @Builder.Default
-  private Set<User> users = newHashSet();
+  @OneToMany(
```
Comment (andricDu) [inline src/main/java/bio/overture/ego/model/entity/Group.java]: :+1:

File: src/main/java/bio/overture/ego/repository/BaseRepository.java [hunk 2]
```diff
@@ -10,5 +10,5 @@
 public interface BaseRepository<T, ID>
     extends PagingAndSortingRepository<T, ID>, JpaSpecificationExecutor {
 
-  Set<T> findAllByIdIn(List<ID> ids);
+  Set<T> findAllByIdIn(Collection<ID> ids);
```
Comment (andricDu) [inline src/main/java/bio/overture/ego/repository/BaseRepository.java]: :+1:

File: src/main/java/bio/overture/ego/repository/GroupRepository.java [hunk 3]
```diff
@@ -16,28 +16,18 @@
...
 
   boolean existsByNameIgnoreCase(String name);
 
-  @EntityGraph(value = "group-entity-with-relationships", type = FETCH)
-  Optional<Group> findGroupById(UUID id);
-
-  Set<Group> findAllByIdIn(List<UUID> groupIds);
-
   @Override
+  @Deprecated
```
Comment (andricDu) [inline src/main/java/bio/overture/ego/repository/GroupRepository.java]: Can you leave a javadoc comment as to why it is deprecated?

File: src/main/java/bio/overture/ego/repository/UserRepository.java [hunk 4]
```diff
@@ -16,28 +16,22 @@
...
-  @EntityGraph(value = "user-entity-with-relationships", type = FETCH)
-  Optional<User> getUserById(UUID id);
-
   boolean existsByEmailIgnoreCase(String email);
 
-  Set<User> findAllByIdIn(List<UUID> userIds);
+  Set<User> findAllByIdIn(Collection<UUID> userIds);
 
   @Override
+  @Deprecated
```
Comment (andricDu) [inline src/main/java/bio/overture/ego/repository/UserRepository.java]: Same deal with the comment for deprecated.

**Comment by rtisma** [comment]
test this please

**Comment by rtisma** [comment]
ok to test

**Pull request merged**